From aa8c7679bab1a9c5d4b992360cf268e9dc65ca47 Mon Sep 17 00:00:00 2001
Message-Id: <aa8c7679bab1a9c5d4b992360cf268e9dc65ca47.1728389216.git.mykyta_poturai@epam.com>
In-Reply-To: <bc35a1559ba18018db00e017ef9359f091274151.1728389216.git.mykyta_poturai@epam.com>
References: <bc35a1559ba18018db00e017ef9359f091274151.1728389216.git.mykyta_poturai@epam.com>
From: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
Date: Tue, 27 Feb 2018 13:47:48 +0200
Subject: [PATCH 05/22] Misc fixes regarding pvz

1. Move GRANT_INVALID_REF definition to pvzif.h as it is going to be
   used from multiple sources.
2. Clarify reserved array size in struct xengsx_req.
3. There is no need to mark ui32FuncID and ui32DevID as unreferenced
   parameters in XenVmmUnmapDevPhysHeap.

Signed-off-by: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
---
 services/system/rogue/common/env/xen/pvzif.h     | 9 ++++++++-
 services/system/rogue/common/env/xen/xen_front.c | 6 ------
 services/system/rogue/common/vmm_type_xen.c      | 2 --
 3 files changed, 8 insertions(+), 9 deletions(-)

diff --git a/services/system/rogue/common/env/xen/pvzif.h b/services/system/rogue/common/env/xen/pvzif.h
index 8b1c4c6..dd5e589 100644
--- a/services/system/rogue/common/env/xen/pvzif.h
+++ b/services/system/rogue/common/env/xen/pvzif.h
@@ -32,6 +32,13 @@
 #include "xen/interface/io/ring.h"
 #include "xen/interface/grant_table.h"
 
+/*
+ * FIXME: usage of grant reference 0 as invalid grant reference:
+ * grant reference 0 is valid, but never exposed to a PV driver,
+ * because of the fact it is already in use/reserved by the PV console.
+ */
+#define GRANT_INVALID_REF	0
+
 /*
  ******************************************************************************
  *                        Example configuration
@@ -100,7 +107,7 @@ struct xengsx_req {
 	uint8_t reserved1[4];
 	union {
 		struct xengsx_map_dev_heap_req map_dev_heap;
-		uint8_t reserved[12];
+		uint8_t reserved[16];
 	} op;
 };
 
diff --git a/services/system/rogue/common/env/xen/xen_front.c b/services/system/rogue/common/env/xen/xen_front.c
index cc1901e..22d9e95 100644
--- a/services/system/rogue/common/env/xen/xen_front.c
+++ b/services/system/rogue/common/env/xen/xen_front.c
@@ -25,12 +25,6 @@
 #include "xen_front.h"
 #include "xen_debug.h"
 
-/*
- * FIXME: usage of grant reference 0 as invalid grant reference:
- * grant reference 0 is valid, but never exposed to a PV driver,
- * because of the fact it is already in use/reserved by the PV console.
- */
-#define GRANT_INVALID_REF	0
 /* timeout in ms to wait for backend to respond */
 #define VGSX_WAIT_BACK_MS	3000
 
diff --git a/services/system/rogue/common/vmm_type_xen.c b/services/system/rogue/common/vmm_type_xen.c
index 9c5d723..522e654 100644
--- a/services/system/rogue/common/vmm_type_xen.c
+++ b/services/system/rogue/common/vmm_type_xen.c
@@ -87,8 +87,6 @@ XenVmmMapDevPhysHeap(IMG_UINT32 ui32FuncID, IMG_UINT32 ui32DevID,
 static PVRSRV_ERROR
 XenVmmUnmapDevPhysHeap(IMG_UINT32 ui32FuncID, IMG_UINT32 ui32DevID)
 {
-	PVR_UNREFERENCED_PARAMETER(ui32FuncID);
-	PVR_UNREFERENCED_PARAMETER(ui32DevID);
 	return gsx_front_unmap(ui32FuncID, ui32DevID);
 }
 
-- 
2.34.1

